server/src/test/java/org/opensearch/gateway/remote/model/RemoteTemplatesMetadataTests.java

```diff
@@ -231,12 +231,22 @@ public static TemplatesMetadata getTemplatesMetadata() {
         return TemplatesMetadata.builder()
             .put(
                 IndexTemplateMetadata.builder("template" + randomAlphaOfLength(3))
+                    .order(1234)
```

Comment:
Can we cover this case in ITs as well?

Reply:
Unit tests should be good to cover this as it basic ser/de issue. 
For ITs we will enable remote publication by default in existing ITs.